### PR TITLE
feat(lambda): encryption key for environment variables

### LINKED ISSUE #10837: [lambda] using encryption configuration kmsKeyArn as parameter not implemented yet.
I have been trying to add an encryption configuration using **KMS** into my Lambda using the parameter `kmsKeyArn` as per the [documentation](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.CfnFunctionProps.html) however it doesn't seem it has been implemented just yet.

Anyone can confirm please?

https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.CfnFunctionProps.html

```      
const lambdaHandler = new lambda.Function(this, lambdaFunction, {
        description: `${lambdaFunction}`,
        timeout: cdk.Duration.seconds(30),
        ...,
        kmsKeyArn???: 
      });
```

---

This is a 📕 documentation issue

### PR DESCRIPTION
**Pull request by alvyn279**
- ARN for KMS key used to encrypt lambda env vars can now be specified in L2 `Function` construct
- fixes #10837

#### Notes
Could we please decorate this PR with the `pr-linter/exempt-readme` label? I don't think this is a big enough feature to add README content.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda/lib/function.ts [hunk 1]
```diff
@@ -282,6 +282,13 @@ export interface FunctionOptions extends EventInvokeConfigOptions {
    * @default false
    */
   readonly allowPublicSubnet?: boolean;
+
+  /**
+   * The ARN of the AWS KMS key that's used to encrypt your function's environment variables.
+   *
+   * @default - AWS Lambda uses a default service key.
+   */
+  readonly kmsKeyArn?: string;
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/lib/function.ts:291]: This needs to be of type [IKey](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-kms.IKey.html).

Several times, I've assumed this to be the kms key for encrypting code and realize it's not until I read the description. We should name this property `environmentEncryption` so as to make it clear.

**Review by nija-at — DISMISSED** [review]
Thanks for submitting the PR. I have one comment below.

File: packages/@aws-cdk/aws-lambda/package.json [hunk 2]
```diff
@@ -122,7 +123,8 @@
     "@aws-cdk/aws-sqs": "0.0.0",
     "@aws-cdk/core": "0.0.0",
     "@aws-cdk/cx-api": "0.0.0",
-    "constructs": "^3.2.0"
+    "constructs": "^3.2.0",
+    "@aws-cdk/aws-kms": "0.0.0"
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/package.json:127]: We usually sort this alphabetically. Can you move this to line 120?

File: packages/@aws-cdk/aws-lambda/lib/function.ts [hunk 3]
```diff
@@ -282,6 +283,13 @@ export interface FunctionOptions extends EventInvokeConfigOptions {
    * @default false
    */
   readonly allowPublicSubnet?: boolean;
+
+  /**
+   * The AWS KMS key that's used to encrypt your function's environment variables.
+   *
+   * @default - AWS Lambda uses a default service key.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/lib/function.ts:290]:
```suggestion
   * @default - AWS Lambda creates and uses an AWS managed customer master key (CMK).
```

**Review by nija-at — CHANGES_REQUESTED** [review]

**Review by nija-at — APPROVED** [review]
Thanks for contributing to the CDK! 😊

**Comment by alvyn279** [comment]
@nija-at Do I need to fix anything (i.e.: the readme linter issue) for the Mergify bot to merge to `master` ?

**Pull request merged**